packages/react-components/react-avatar/src/utils/partitionAvatarGroupItems.ts

```diff
@@ -0,0 +1,36 @@
+export type PartitionAvatarGroupItemsOptions<T> = {
+  items: readonly T[];
+  layout?: 'spread' | 'stack' | 'pie';
+  maxInlineItems?: number;
+};
+
+/**
+ * Get the inline items and overflowing items based on the array of AvatarGroupItems needed for AvatarGroup.
+ *
+ * @param options - Configure the partition options
+ *
+ * @returns Two arrays of AvatarGroupItems splitted between the inline items and the overflowing items
+ * based on maxItems.
+ *
+ * NOTE: Each item in the array must be a single AvatarGroupItem. If an item in the array contains a fragment with
+ * two avatars, the items will not be partitioned correctly.
```

Comment:
This isn't entirely correct now, since this works on any array of data, and not just avatar group items. It's probably better to just remove it?